src/plone/restapi/deserializer/blocks.py

```diff
@@ -12,20 +12,34 @@
 from zope.publisher.interfaces.browser import IBrowserRequest
 
 
-def path2uid(context, path):
+def path2uid(context, portal, href):
     # unrestrictedTraverse requires a string on py3. see:
     # https://github.com/zopefoundation/Zope/issues/674
-    if not isinstance(path, str):
```

Comment:
@cekk are you observing this "unrestrictedTraverse requires a string on py3. see:" comment later in the new code? I cannot see the check.

Reply:
I'd like to understand as well why we switched from unrestrictedTraverse to getPhysicalPath

Reply:
@sneridagh  it was an old comment made by who started this feature.
@tisto first of all (rows 20-30) i'm trying to get the right object path because i'm handling full urls (that could be without the site name) and paths. I replace the portal_url from the given string and append the portal_path to. have the right string to pass to unrestrictedTraverse to check if the object exists.
More clear now?